File: Best Time to Buy and Sell Stock/Best Time to Buy And Sell A Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minimumSoFar = prices[0];
        int ans = 0;
        for(int i = 1; i < prices.size(); i++) {
            int currentProfit = prices[i] - minimumSoFar;
            ans = max(ans, currentProfit);
            minimumSoFar = min(minimumSoFar, prices[i]);
        }
        return ans;
    }
};

File: Best Time to Buy and Sell Stock/Best Time to Buy And Sell A Stock.py
class Solution(object):
    def maxProfit(self, prices):
      result=0
      maxValue=prices[0]
      minValue=0
#   print("Hello from a function")
      for i in range(1,len(prices)):
       if maxValue>prices[i]:
        maxValue=prices[i]
       else:
        minValue=prices[i]-maxValue
        if minValue>result:
         result=minValue
      return result